contents/docs/libraries/next-js/_snippets/app-router-server-setup.mdx

```diff
@@ -0,0 +1,44 @@
+For the app router, we can initialize the `posthog-node` SDK once with a `PostHogClient` function, and import it into files.
+
+This enables us to send events and fetch data from PostHog on the server – without making client-side requests.
+
+```js
+// app/posthog.js
+import { PostHog } from 'posthog-node'
+
+export default function PostHogClient() {
+  const posthogClient = new PostHog(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
+    host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
+    flushAt: 1,
+    flushInterval: 0
+  })
+  return posthogClient
+}
+```
+
+> **Note:** Because our server-side `posthog-node` initializations are short-lived, we set `flushAt` to `1` and `flushInterval` to `0`. `flushAt` sets how many capture calls we should flush the queue (in one batch). `flushInterval` sets how many milliseconds we should wait before flushing the queue. Setting them to the lowest number ensures events are sent immediately and not batched. We also need to call `await posthog.shutdown()` once done.
```

Comment:
"Because our server-side `posthog-node` initializations are short-lived" makes it sound like the problem is us, instead of the fact that serverless functions are shortlived. Lets rephrase to emphasize that its a serverless thing and not a posthog thing

also, why do we need to change flush if we're alerady calling shutdown() ? Wont it have the same effect?

Reply:
I'll clarify what's there now, but I didn't change this from what's already there. 

I assume we need to flush immediately and await shutdown to both trigger the flush and make sure it completes.